### PR TITLE
[Enhancement] Make region is optional in iceberg rest catalog vended credentials

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## Why I'm doing:
For Polaris's vended-credentials, it will not return aws's region, we can't treat it as required.

## What I'm doing:
Fix it and also add doc.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0

### CODE REVIEW

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 1]
```diff
@@ -258,18 +259,50 @@ Description: The type of metastore that you use for your Iceberg cluster. Set th
...
+Description: The URI of the REST service endpoint. Example: `https://api.tabular.io/ws`. 
 
-###### iceberg.catalog.credential
+###### iceberg.catalog.security
 
-Required: Yes
-Description: The authentication information of the Tabular service.
+Required: No
+
+Description: The type of security to use (default: `NONE`). `OAUTH2` requires either a `token` or `credential`. Example: `OAUTH2`.
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:268]:
```suggestion
Description: The type of authorization protocol to use (default: `NONE`). `OAUTH2` requires either a `token` or `credential`. Example: `OAUTH2`.
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 2]
```diff
@@ -258,18 +259,50 @@ Description: The type of metastore that you use for your Iceberg cluster. Set th
...
+
+###### iceberg.catalog.oauth2.token
+
+Required: No
+
+Description: The bearer token used for interactions with the server. A `token` or `credential` is required for `OAUTH2` security. Example: `AbCdEf123456`.
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:274]:
```suggestion
Description: The bearer token used for interactions with the server. A `token` or `credential` is required for `OAUTH2` authorization protocol. Example: `AbCdEf123456`.
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 3]
```diff
@@ -258,18 +259,50 @@ Description: The type of metastore that you use for your Iceberg cluster. Set th
...
+
+###### iceberg.catalog.oauth2.credential
+
+Required: No
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:278]:
```suggestion
Required: No

```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 4]
```diff
@@ -258,18 +259,50 @@ Description: The type of metastore that you use for your Iceberg cluster. Set th
...
+Description: The credential to exchange for a token in the OAuth2 client credentials flow with the server. A `token` or `credential` is required for `OAUTH2` security. Example: `AbCdEf123456`.
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:279]:
```suggestion
Description: The credential to exchange for a token in the OAuth2 client credentials flow with the server. A `token` or `credential` is required for `OAUTH2` authorization protocol. Example: `AbCdEf123456`.
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 5]
```diff
@@ -258,18 +259,50 @@ Description: The type of metastore that you use for your Iceberg cluster. Set th
...
+
+###### iceberg.catalog.oauth2.scope
+
+Required: No
+
+Description: Scope to be used when communicating with the REST Catalog. Applicable only when using `credential`.
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:285]:
```suggestion
Description: Scope to be used when communicating with the REST Catalog. Applicable only when `credential` is used.
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 6]
```diff
@@ -258,18 +259,50 @@ Description: The type of metastore that you use for your Iceberg cluster. Set th
...
+
+Required: No
+
+Description: The endpoint to retrieve access token from OAuth2 Server.
+
+###### iceberg.catalog.vended-credentials-enabled
+
+Required: No
+
+Description: Support querying objects under nested namespace. Defaults to `true`.
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:297]:
```suggestion
Description: Whether to support querying objects under nested namespace. Default: `true`.
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 7]
```diff
@@ -283,6 +316,34 @@ PROPERTIES
...
+
+```sql
+CREATE EXTERNAL CATALOG smith_polaris 
+PROPERTIES (   
+    "iceberg.catalog.uri"  = "http://localhost:8181/api/catalog",   
+    "type"  =  "iceberg",   
+    "iceberg.catalog.type"  =  "rest",   
+    "iceberg.catalog.warehouse" = "starrocks_catalog",
+    "iceberg.catalog.security" = "oauth2",
+    "iceberg.catalog.oauth2.credential" = "f7357cfbe9583aa1:7b90fc77e4fbe288a5c41eaa05da3a3b",
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:329]:
```suggestion
    "iceberg.catalog.oauth2.credential" = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 8]
```diff
@@ -283,6 +316,34 @@ PROPERTIES
     "iceberg.catalog.warehouse" = "sandbox"
 );
 ```
+The following example creates an Iceberg catalog named `smith_polaris` that uses Polaris as metastore:
+
+```sql
+CREATE EXTERNAL CATALOG smith_polaris 
+PROPERTIES (   
+    "iceberg.catalog.uri"  = "http://localhost:8181/api/catalog",
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:324]:
```suggestion
    "iceberg.catalog.uri"  = "http://xxx.xx.xx.xxx:8181/api/catalog",   
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 9]
```diff
@@ -283,6 +316,34 @@ PROPERTIES
     "iceberg.catalog.warehouse" = "sandbox"
 );
 ```
+The following example creates an Iceberg catalog named `smith_polaris` that uses Polaris as metastore:
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:319]:
```suggestion
The following example creates an Iceberg catalog named `iceberg_polaris` that uses Polaris as metastore:
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 10]
```diff
@@ -283,6 +316,34 @@ PROPERTIES
...
+
+```sql
+CREATE EXTERNAL CATALOG smith_polaris
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:322]:
```suggestion
CREATE EXTERNAL CATALOG iceberg_polaris 
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 11]
```diff
@@ -283,6 +316,34 @@ PROPERTIES
...
+    "type"  =  "iceberg",   
+    "iceberg.catalog.type"  =  "rest",   
+    "iceberg.catalog.warehouse" = "starrocks_catalog",
+    "iceberg.catalog.security" = "oauth2",
+    "iceberg.catalog.oauth2.credential" = "f7357cfbe9583aa1:7b90fc77e4fbe288a5c41eaa05da3a3b",
+    "iceberg.catalog.oauth2.scope"='PRINCIPAL_ROLE:ALL'
+ );
+
+# `ns1.ns2.tpch_namespace` is a nested namespace
+create table smith_polaris.`ns1.ns2.tpch_namespace`.tbl (c1 string);
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:334]:
```suggestion
create table iceberg_polaris.`ns1.ns2.tpch_namespace`.tbl (c1 string);
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 12]
```diff
@@ -283,6 +316,34 @@ PROPERTIES
...
+
+mysql> select * from smith_polaris.`ns1.ns2.tpch_namespace`.tbl;
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:336]:
```suggestion
mysql> select * from iceberg_polaris.`ns1.ns2.tpch_namespace`.tbl;
```

File: docs/en/data_source/catalog/iceberg_catalog.md (hunk 1 above)
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:268]:
```suggestion
Description: The type of authorization protocol to use. Default: `NONE`. Valid value: `OAUTH2`, which requires either a `token` or `credential`.
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 13]
```diff
@@ -258,18 +259,50 @@ Description: The type of metastore that you use for your Iceberg cluster. Set th
...
-Description: The authentication information of the Tabular service.
+Required: No
+
+Description: The type of authorization protocol to use. Default: `NONE`. Valid value: `OAUTH2`, which requires either a `token` or `credential`.
+
+###### iceberg.catalog.oauth2.token
+
+Required: No
+
+Description: The bearer token used for interactions with the server. A `token` or `credential` is required for `OAUTH2` security. Example: `AbCdEf123456`.
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:274]:
```suggestion
Description: The bearer token used for interactions with the server. A `token` or `credential` is required for `OAUTH2` authorization protocol. Example: `AbCdEf123456`.
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 14]
```diff
@@ -258,18 +259,50 @@ Description: The type of metastore that you use for your Iceberg cluster. Set th
...
+
+###### iceberg.catalog.oauth2.token
+
+Required: No
+
+Description: The bearer token used for interactions with the server. A `token` or `credential` is required for `OAUTH2` authorization protocol. Example: `AbCdEf123456`.
+
+###### iceberg.catalog.oauth2.credential
+
+Required: No
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:278]:
```suggestion
Required: No

```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 15]
```diff
@@ -258,18 +259,51 @@ Description: The type of metastore that you use for your Iceberg cluster. Set th
...
+
+Description: The credential to exchange for a token in the OAuth2 client credentials flow with the server. A `token` or `credential` is required for `OAUTH2` security. Example: `AbCdEf123456`.
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:280]:
```suggestion
Description: The credential to exchange for a token in the OAuth2 client credentials flow with the server. A `token` or `credential` is required for `OAUTH2` authorization protocol. Example: `AbCdEf123456`.
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 16]
```diff
@@ -258,18 +259,51 @@ Description: The type of metastore that you use for your Iceberg cluster. Set th
...
+
+Required: No
+
+Description: The credential to exchange for a token in the OAuth2 client credentials flow with the server. A `token` or `credential` is required for `OAUTH2` authorization protocol. Example: `AbCdEf123456`.
+
+###### iceberg.catalog.oauth2.scope
+
+Required: No
+
+Description: Scope to be used when communicating with the REST Catalog. Applicable only when using `credential`.
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:286]:
```suggestion
Description: Scope to be used when communicating with the REST Catalog. Applicable only when `credential` is used.
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 17]
```diff
@@ -258,18 +259,51 @@ Description: The type of metastore that you use for your Iceberg cluster. Set th
...
+
+Required: No
+
+Description: The endpoint to retrieve access token from OAuth2 Server.
+
+###### iceberg.catalog.vended-credentials-enabled
+
+Required: No
+
+Description: Support querying objects under nested namespace. Defaults to `true`.
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:298]:
```suggestion
Description: Whether to support querying objects under nested namespace. Default: `true`.
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 18]
```diff
@@ -283,6 +317,34 @@ PROPERTIES
     "iceberg.catalog.warehouse" = "sandbox"
 );
 ```
+The following example creates an Iceberg catalog named `smith_polaris` that uses Polaris as metastore:
+
+```sql
+CREATE EXTERNAL CATALOG smith_polaris 
+PROPERTIES (   
+    "iceberg.catalog.uri"  = "http://localhost:8181/api/catalog",
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:325]:
```suggestion
    "iceberg.catalog.uri"  = "http://xxx.xx.xx.xxx:8181/api/catalog", 
```

File: docs/en/data_source/catalog/iceberg_catalog.md [hunk 19]
```diff
@@ -283,6 +317,34 @@ PROPERTIES
...
+
+```sql
+CREATE EXTERNAL CATALOG smith_polaris 
+PROPERTIES (   
+    "iceberg.catalog.uri"  = "http://xxx.xx.xx.xxx:8181/api/catalog", 
+    "type"  =  "iceberg",   
+    "iceberg.catalog.type"  =  "rest",   
+    "iceberg.catalog.warehouse" = "starrocks_catalog",
+    "iceberg.catalog.security" = "oauth2",
+    "iceberg.catalog.oauth2.credential" = "f7357cfbe9583aa1:7b90fc77e4fbe288a5c41eaa05da3a3b",
```
Comment (EsoragotoSpirit) [inline docs/en/data_source/catalog/iceberg_catalog.md:330]:
```suggestion
    "iceberg.catalog.oauth2.credential" = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
```

**Review by EsoragotoSpirit — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-3.4

**Pull request merged**